### ISSUE TITLE
Inconsistency in traceability report

### ISSUE THREAD

**Issue by ayankuma** [itsallcode/openfasttrace#401]
# Description
The traceability report is inconsistent while using MD heading style vs RST heading style. The inconsistency is seen only with the first item

## Steps to Reproduce
Consider the following tmp.rst
```
This is the main heading                                                                 
========================                                                                 
                                                                                         
The following are the requirements related to domain creation.                  
                                                                                         
Item 1 heading                                                                           
##############                                                                           
                                                                                         
`XenSSR~item1~1`                                                                         
                                                                                         
Description:                                                                             
Item 1 descrip.                                                                          
                                                                                         
Rationale:                                                                               
Item 1 rationale.                                                                        
                                                                                         
Item 2 heading                                                                           
##############                                                                           
                                                                                         
`XenSSR~item2~1`                                                                         
                                                                                         
Description:                                                                             
Item 2 descrip.                                                                          
                                                                                         
Rationale:                                                                               
Item 2 rationale.                                                                        
                                                                                         
Item 3 heading                                                                           
##############                                                                           
                                                                                         
`XenSSR~item3~1`                                                                         
                                                                                         
Description:                                                                             
Item 3 descrip.                                                                          
                                                                                         
Rationale:                                                                               
Item 3 rationale.                                                                        

```

The traceability report looks like. This is correct and what we expect.
![Screenshot 2024-03-05 at 19 31 48](https://github.com/itsallcode/openfasttrace/assets/91470115/e624a36a-2bfe-47bb-a08c-ae4a9596a047)

However, when we use RST heading styles ie (the below file)
```
This is the main heading                                                                 
========================                                                                 
                                                                                         
The following are the requirements related to domain creation.                  
                                                                                         
Item 1 heading                                                                           
--------------
                                                                                         
`XenSSR~item1~1`                                                                         
                                                                                         
Description:                                                                             
Item 1 descrip.                                                                          
                                                                                         
Rationale:                                                                               
Item 1 rationale.                                                                        
                                                                                         
Item 2 heading                                                                           
--------------
                                                                                         
`XenSSR~item2~1`                                                                         
                                                                                         
Description:                                                                             
Item 2 descrip.                                                                          
                                                                                         
Rationale:                                                                               
Item 2 rationale.                                                                        
                                                                                         
Item 3 heading                                                                           
--------------
                                                                                         
`XenSSR~item3~1`                                                                         
                                                                                         
Description:                                                                             
Item 3 descrip.                                                                          
                                                                                         
Rationale:                                                                               
Item 3 rationale.                                                                        

```

The traceability report looks like this 
![Screenshot 2024-03-05 at 19 35 05](https://github.com/itsallcode/openfasttrace/assets/91470115/e6b72c52-f7a8-4214-9245-5f14d5de842b)

## Expected behavior
The issue is that for the first item, it renders the heading. We don't want this. Instead we expect that it should render the tags as seen for item2 and item3.
In other words with RST header style, the rendering should be similar to the markdown header style.

# Environment
- OFT: This is the latest master.
- OS: [Ubuntu 20.04]
